packages/fx-core/src/core/middleware/utils/migrationContext.ts

```diff
@@ -106,4 +106,28 @@ export class MigrationContext {
   async cleanTeamsfx(): Promise<void> {
     await fs.remove(path.join(this.projectPath, teamsfxFolder));
   }
+
+  async readStateFile(filePath: string): Promise<any> {
+    const filepath = path.join(this.projectPath, filePath);
+    if (await fs.pathExists(filepath)) {
```

Comment:
as internal utility function, you're able remove some checks and let consumer to ensure the path exists. Because you list files first then read the files, so the file will always exist.